test/integration/apiserver/anonymous/anonymous_test.go

```diff
@@ -0,0 +1,332 @@
+	t.Log("Testing anonymous authenticator with authentication config")
+
+	testCases := []struct {
+		desc            string
+		featureGates    map[featuregate.Feature]bool
+		authConfig      string
+		additionalFlags []string
+		assertErrFn     func(t *testing.T, errorToCheck error)
+		assertFn        func(t *testing.T, server kubeapiserverapptesting.TestServer)
+	}{
+		{
+			desc: "StructuredAuthenticationConfiguration disabled",
+			featureGates: map[featuregate.Feature]bool{
+				features.StructuredAuthenticationConfiguration: false,
+			},
+			authConfig: `
+apiVersion: apiserver.config.k8s.io/v1beta1
+kind: AuthenticationConfiguration
+anonymous:
+  enabled: true
+  conditions:
+  - path: "/livez"
+`,
+			assertErrFn: func(t *testing.T, errorToCheck error) {
+				assert.EqualError(t, errorToCheck, "failed to validate ServerRunOptions: set --feature-gates=StructuredAuthenticationConfiguration=true to use authentication-config file")
+			},
+		},
+		{
+			desc: "AnonymousAuthConfigurableEndpoints disabled",
+			featureGates: map[featuregate.Feature]bool{
+				features.StructuredAuthenticationConfiguration: true,
+				features.AnonymousAuthConfigurableEndpoints:    false,
+			},
+			authConfig: `
+apiVersion: apiserver.config.k8s.io/v1beta1
+kind: AuthenticationConfiguration
+anonymous:
+  enabled: true
+  conditions:
+  - path: "/livez"
+`,
+			assertErrFn: func(t *testing.T, errorToCheck error) {
+				assert.EqualError(t, errorToCheck, "anonymous: Forbidden: anonymous is not supported when when AnonymousAuthConfigurableEnpoints feature gate is disabled")
+			},
+		},
+		{
+			desc: "invalid flags",
+			featureGates: map[featuregate.Feature]bool{
+				features.StructuredAuthenticationConfiguration: true,
+				features.AnonymousAuthConfigurableEndpoints:    true,
+			},
+			authConfig: `
+apiVersion: apiserver.config.k8s.io/v1beta1
+kind: AuthenticationConfiguration
+anonymous:
+  enabled: true
+  conditions:
+  - path: "/livez"
+`,
+			assertErrFn: func(t *testing.T, errorToCheck error) {
+				assert.EqualError(t, errorToCheck, "anonymous: Forbidden: --anonynous-auth flag cannot be set when anonymous field is configured in authentication configuration file")
+			},
+			additionalFlags: []string{"--anonymous-auth=true"},
+		},
+		{
+			desc: "invalid config - anonymous enabled unset conditions set",
+			featureGates: map[featuregate.Feature]bool{
+				features.StructuredAuthenticationConfiguration: true,
+				features.AnonymousAuthConfigurableEndpoints:    true,
+			},
+			authConfig: `
+apiVersion: apiserver.config.k8s.io/v1beta1
+kind: AuthenticationConfiguration
+anonymous:
+  conditions:
+  - path: "/livez"
+`,
+			assertErrFn: func(t *testing.T, errorToCheck error) {
+				assert.EqualError(t, errorToCheck, "anonymous.conditions: Invalid value: []apiserver.AnonymousAuthCondition{apiserver.AnonymousAuthCondition{Path:\"/livez\"}}: enabled should be set to true when conditions are defined")
```

Comment:
```suggestion
				assert.EqualError(t, errorToCheck, "anonymous.conditions: Invalid value: []apiserver.AnonymousAuthCondition{apiserver.AnonymousAuthCondition{Path:\"/livez\"}}: "enabled" should be set to "true" when conditions are defined")
```

key/values should be ideally quoted.

Reply:
updated. Thanks!